pokemongo_bot/cell_workers/incubate_eggs_worker.py

```diff
@@ -0,0 +1,65 @@
+from utils import distance, format_dist
+from pokemongo_bot.human_behaviour import sleep
+from pokemongo_bot import logger
+
+
+class IncubateEggsWorker(object):
+    def __init__(self, bot):
+        self.api = bot.api
+        self.config = bot.config
+        self.bot = bot
+        # self.position = bot.position
+
+    def work(self):
+        if not self.config.hatch_eggs:
+            return
+
+        self.api.get_inventory()
+        response_dict = self.api.call()
+        inv = {}
+        incubators = []
+        eggs = []
+
+        try:
+            inv = reduce(dict.__getitem__, [
+                "responses", "GET_INVENTORY", "inventory_delta", "inventory_items"], response_dict)
+        except KeyError:
+            pass
+        else:
+            for inv_data in inv:
+                inv_data = inv_data.get("inventory_item_data", {})
+
+                if "egg_incubators" in inv_data:
+                    for incubator in inv_data.get("egg_incubators", {}).get("egg_incubator", []):
+                        incubators.append({"id":incubator.get("item_id", -1), "used":False})
+
```

Comment:
You can check if the incubator is being used here.  It will have 'pokemon_id' set if there is an egg in it.  'pokemon_id' matches 'id' from an egg.